nixos/modules/services/networking/hylafaxp.nix

```diff
@@ -0,0 +1,262 @@
+        #    the default entry.
+        #						Guaranteed Reproducible Area
+        # Name			Abbrev	Width	Height	Width	Height	Top	Left
+        ISO A3			A3	14030	19840	13200	 18480	472	345
+        ISO A4			A4	 9920	14030	 9240	 13200	472	345
+        ISO A5			A5	 7133	 9921	 6455	  9236	472	345
+        ISO A6			A6	 5055	 6991	 4575	  6508	472	345
+        ISO B4			B4	12048	17196	11325	 16010	472	345
+        North American Letter	NA-LET	10200	13200	 9240	 12400	472	345
+        American Legal		US-LEG	10200	16800	 9240	 15775	472	345
+        American Ledger		US-LED	13200	20400	11946	 19162	472	345
+        American Executive	US-EXE	 8700	12600	 7874	 11835	472	345
+        Japanese Letter		JP-LET	 8598	12141	 7600	 10200	900	400
+        Japanese Legal		JP-LEG	12141	17196	11200	 15300	900	400
+        #
+        default			A4	 9920	14030	 9240	 13200	472	345
+      '';
+      description = ''
+        Page sizes as defined in /etc/hylafax/pagesizes.
+        Defaults to A4.
+      '';
+    };
+
+    services.hylafaxp.faxDispatch = mkOption {
+      type = types.lines;
+      default = "";
+      example = ''
+        SENDTO=FaxMaster;
+        FILETYPE=tif;
+      '';
+      description = ''
+        faxrcvd's facsimile dispatching script at /var/spool/hylafax/etc/FaxDispatch.
+      '';
+    };
+
+    services.hylafaxp.hfaxd = {
+      enable = mkOption {
+        type = types.bool;
+        default = false ;
+        description = ''
+          Whether to enable hfaxd, HylaFAX+ client-server protocol server.
+        '';
+      };
+      port = mkOption {
+        type = types.int;
+        default = 4559 ;
+        description = ''
+          hfaxd's port.
+        '';
+      };
+      hosts = mkOption {
+        type = types.lines;
+        default = ''
+          localhost
+          127.0.0.1
+        '';
+        description = ''
+          The client access control list at /var/spool/hylafax/etc/hosts.hfaxd.
+          Run faxadduser to add hosts, users and passwords.
+          Be sure to copy the results to configuration.nix!
+        '';
+      };
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf config.services.hylafaxp.enable {
+    environment.systemPackages = with pkgs; [ hylafaxp ];
+
+    users.extraUsers = [
+      {
+        name = "uucp";
+        uid = config.ids.uids.uucp;
+        group = "uucp";
+      }
+    ];
+
+    systemd = {
+      # the services refork as uucp on their own
+
+      targets.hylafaxp = {
+        description = "HylaFAX+ Server";
+        after = [ "network.target" ];
+        wantedBy = [ "multi-user.target" ];
+      };
+
+      services =
+      let
+        mkFaxgettyService = dev:
+          assert dev != "";
+          nameValuePair "hylafaxp-faxgetty-${dev}" {
+            description = "faxgetty ${dev}: HylaFAX+ front-door process";
+            after = optionals (dev!="dummy") [ "hylafaxp-setup.service" "hylafaxp-faxq.service" "${dev}.device" ];
```

Comment:
@Mic92 This one sucks but I'm not sure how to clean it up.

I'd like to condition this entire let..in.. construct by config.services.hylafaxp.devs and move the error message* to the setupscript.

It's quite ugly and convoluted in an already complex structure but I don't know how to do the condition so I ended up recycling the unit to display the error letting the user know they should run faxsetup and update config.services.hylafaxp.devs accordingly so the unit won't be a total waste.

Similarly, there are also a few setupScript config.services.hylafaxp.devs if statements I'm not particularly happy about.